### ISSUE TITLE
dice problem

### ISSUE THREAD

**Issue by xizhedong** [sotiraslab/AgileFormer#7]
这个dice计算有点问题吧。
![image](https://github.com/sotiraslab/AgileFormer/assets/69316039/89d1c761-1a43-4d2a-be24-6d87861b2f8f)
这是开源框架[monai](https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/meandice.py)计算方式
![image](https://github.com/sotiraslab/AgileFormer/assets/69316039/3cc3e8d2-144e-473b-8d52-bd80e8a586f5)

**Comment by peijieq** [comment]
We followed the dice computation as in SwinUNet (https://github.com/HuCaoFighting/Swin-Unet) and nnFormer (https://github.com/282857341/nnFormer) to ensure a fair comparison. We are aware of the issue in the dice computation in the previous literature (see https://github.com/McGregorWwww/UDTransNet/issues/4). For the purpose of a fair comparison, we have to keep consistency.

**Comment by xizhedong** [comment]
[nnformer/inference_synapse.py](https://github.com/282857341/nnFormer/blob/main/nnformer/inference_synapse.py)dice计算与swin-unet并不一样，[UDTransNet](https://github.com/McGregorWwww/UDTransNet/issues/4)，有了自己的benchmark,且明显高于实际,如果按照swin-unet的计算方式，如果图片背景居多，可能导致dcie出现明显的偏差。

**Comment by peijieq** [comment]
That's absolutely correct, and I agree with that. What I meant (following the same dice computation in SwinUNet and nnFormer) is that for the 2D experiments, we followed the dice computation method from SwinUNet, while for the 3D experiments, we followed the dice computation method from nnFormer. But as I said, that's for the purpose of a fair comparison with the performance reported by previous literature. We're certainly happy to recompute the metrics using a more "rigorous" way of computing Dice.
